Modules/_io/bytesio.c

```diff
@@ -404,15 +405,21 @@ read_bytes(bytesio *self, Py_ssize_t size)
     if (size > 1 &&
         self->pos == 0 && size == PyBytes_GET_SIZE(self->buf) &&
         self->exports == 0) {
-        self->pos += size;
         return Py_NewRef(self->buf);
     }
 
     output = PyBytes_AS_STRING(self->buf) + self->pos;
-    self->pos += size;
     return PyBytes_FromStringAndSize(output, size);
 }
 
+static PyObject *
+read_bytes(bytesio *self, Py_ssize_t size)
+{
+    PyObject *bytes = peek_bytes(self, size);
+    self->pos += size;
```

Comment:
You know what? It's an opportunity to fix another bug! If PyBytes_FromStringAndSize() fails, so if bytes is NULL, you should not update ``self->pos``. Can you please fix that?